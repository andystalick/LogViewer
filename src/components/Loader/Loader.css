.loader {
  padding: 0.25rem 0.25rem;
  font-size: 0.9rem;
  background-color: #fff;
  progress {
    width: 100%;
  }
}
