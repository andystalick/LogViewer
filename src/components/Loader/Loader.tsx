interface LoaderProps {
  loaded: number;
  total: number;
}

import './Loader.css';

const Loader: React.FC<LoaderProps> = (props) => {
  const { loaded, total } = props;

  return (
    <div className="loader" data-testid="loader">
      {`Loaded ${loaded} of ${total} items`}
      <progress max={total} value={loaded}></progress>
    </div>
  );
};

export default Loader;
