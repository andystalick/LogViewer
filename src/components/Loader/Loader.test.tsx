import { expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import Loader from './Loader';

test('snapshot test', async () => {
  render(<Loader loaded={50} total={100} />);
  expect(screen.getByTestId('loader')).toMatchSnapshot();
});
