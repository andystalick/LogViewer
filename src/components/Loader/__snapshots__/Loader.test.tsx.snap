// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test 1`] = `
<div
  class="loader"
  data-testid="loader"
>
  Loaded 50 of 100 items
  <progress
    max="100"
    value="50"
  />
</div>
`;
